### PR TITLE
[wip]secret-syncer: ensure the target namespace exists

### PR DESCRIPTION
**Pull request by hongkailiu**
Follow up https://coreos.slack.com/archives/CHY2E1BL4/p1613674179255500

/cc @alvaroaleman @petr-muller

### CODE REVIEW

File: pkg/controller/secretsyncer/secretsyncer.go [hunk 1]
```diff
@@ -188,6 +188,19 @@ func (r *reconciler) mirrorSecret(ctx context.Context, log *logrus.Entry, cluste
 		}
 		data[k] = source.Data[k]
 	}
+
+	if err := c.Get(ctx, types.NamespacedName{Name: to.Namespace}, &corev1.Namespace{}); err != nil {
```
Comment (alvaroaleman) [inline pkg/controller/secretsyncer/secretsyncer.go:192]: Can we instead add a config option to limit the target clusters? Users are not supposed to use this to create namesapces. The secret in question is only needed for app.ci

Comment (hongkailiu) [reply pkg/controller/secretsyncer/secretsyncer.go:192]: Ack. Will do something like cluster group.

Comment (hongkailiu) [reply pkg/controller/secretsyncer/secretsyncer.go:192]: Hu.
It has been implemented.
https://github.com/openshift/ci-tools/blob/b0351c25136f3e2695d5f2364dada541010230fe/pkg/controller/secretsyncer/config/config.go#L42

Comment (hongkailiu) [reply pkg/controller/secretsyncer/secretsyncer.go:192]: /close

https://github.com/openshift/release/pull/16036

**Pull request closed without merging**